hack/generator/spec/crossplane.md

```diff
@@ -0,0 +1,71 @@
+# Thoughts on performing provider-azure code generation for Crossplane
+
+The patterns identified below were extrapolated from [the Crossplane VNET spec](https://github.com/crossplane/provider-azure/blob/master/config/crd/network.azure.crossplane.io_virtualnetworks.yaml).
+
+## General notes
+
+### Crossplane has the following static "special" properties
+  1. ClaimRef [Deprecated]
+  2. ClassRef [Deprecated]
+  3. DeletionPolicy
+  4. ProviderConfigRef
+  5. ProviderRef [Deprecated]
+  6. ReclaimPolicy [Deprecated]
+  7. writeConnectionSecretToRef
+
+Since these are statically shaped they should be relatively easy to add with Crossplane specific generator pipeline stages.
+
+### Parent references in Crossplane are really a discriminated union of 3 options:
+1. XName (e.g. ResourceGroupName)
+2. XNameRef (e.g. ResourceGroupNameRef)
+3. XNameSelector (e.g. ResourceGroupNameSelector)
+
+We probably will eventually want to understand how `XName` differs from `XNameRef` (I think I understand the `XNameSelector`). 
```

Comment:
We discussed this a little out of band, but the idea is:

* `X` - An actual field that exists in the Azure API. Can be set to any string you'd post to the Azure API.
* `XRef` - A reference to a named Kubernetes object of a preordained kind. Used to resolve a value for `X`.
* `XSelector` - A selector used to set `XRef`.

Note that these values cascade down from the selector, to the ref, to the underlying field. If the ref is already set the selector is ignored, and if the field is already set the ref is ignored.

In practice I think it's fair to call these parent references, but note that they don't _have_ to be. We call these "cross resource references" - the idea is that where they exist they can be used to derive a field value by inspecting another resource. Note also that the derived field doesn't necessarily have the suffix `Name` - that just happens to be the convention in the Azure API.

Reply:
Some breadcrumbs for the logic we use today; https://github.com/crossplane/provider-azure/blob/64dc01e/apis/cache/v1beta1/references.go#L29

Reply:
> Note that these values cascade down from the selector, to the ref, to the underlying field. If the ref is already set the selector is ignored, and if the field is already set the ref is ignored.

@ngez just clarifying, this means that if only the `XSelector` is specified, the `XRef` and `X` fields will be visible by the customer when they examine the object? Or will they be empty?